File: src/main/java/me/andre111/d20server/template/PlaceholderRollInline.java
package me.andre111.d20server.template;

import java.util.List;

import me.andre111.d20common.model.chat.ChatEntry.TriggeredContent;
import me.andre111.d20common.model.profile.Profile;
import me.andre111.d20common.scripting.Context;
import me.andre111.d20common.scripting.ScriptException;
import me.andre111.d20common.scripting.expression.DiceRoll;
import me.andre111.d20common.scripting.expression.Expression;
import me.andre111.d20common.scripting.expression.Parser;
import me.andre111.d20common.scripting.expression.Result;
import me.andre111.d20server.util.RollFormatter;

public class PlaceholderRollInline extends Placeholder {
	private final Parser parser = new Parser();

	@Override
	public String parse(Profile profile, String input, List<DiceRoll> diceRolls, List<TriggeredContent> triggeredContent) throws ScriptException {
		// parse roll and execute
		Result result = null;
		Exception exception = null;
		try {
			Expression expr = parser.parse(input);
			result = expr.eval(new Context(profile, profile.getMap(), null));
		} catch(Exception e) {
			exception = e;
		}
		if(result != null) diceRolls.addAll(result.getDiceRolls());
		
		return RollFormatter.formatInlineDiceRoll(input, result, exception != null ? exception.getMessage() : null);
	}
}

File: core/server/template/placeholder-roll-inline.js
import { Placeholder } from './placeholder.js';

import { Parser } from '../../common/scripting/expression/parser.js';
import { Context } from '../../common/scripting/context.js';
import { EntityManagers } from '../../common/entity/entity-managers.js';
import { RollFormatter } from '../util/roll-formatter.js';

export class PlaceholderRollInline extends Placeholder {
    parser = new Parser();

    parse(profile, input, diceRolls, triggeredContent) {
        // parse roll and execute
        var result = null;
        var error = null;
        try {
            const expr = this.parser.parse(input);
            result = expr.eval(new Context(profile, EntityManagers.get('map').find(profile.getCurrentMap()), null));
        } catch(e) {
            error = e;
        }
        if(result) {
            for(const diceRoll of result.getDiceRolls()) {
                diceRolls.push(diceRoll);
            }
        }

        return RollFormatter.formatInlineDiceRoll(input, result, error);
    }
}
